Snowflake.Data.Tests/Mock/MockS3Client.cs

```diff
@@ -0,0 +1,122 @@
+﻿/*
+ * Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+ */
+
+using Amazon.S3;
+using Amazon.S3.Model;
+using System.Threading;
+using System.Threading.Tasks;
+using Snowflake.Data.Core.FileTransfer.StorageClient;
+using System;
+using System.IO;
+using System.Text;
+using System.Net.Http;
+using System.Net;
+
+namespace Snowflake.Data.Tests.Mock
+{
+    class MockAmazonS3Client : AmazonS3Client
+    {
+        // Mock status codes for S3 requests
+        internal const string AwsStatusOk = "OK";
+        internal const string AwsStatusError = "AWS_ERROR";
+
+        // Mock S3 data for FileHeader
+        internal const string AmzIV = "MOCK_AMZ_IV";
+        internal const string AmzKey = "MOCK_AMZ_KEY";
+        internal const string AmzMatdesc = "MOCK_AMZ_MATDESC";
+        internal const string SfcDigest = "MOCK_SFC_DIGEST";
+
+        // Mock error message for S3 errors
+        internal const string S3ErrorMessage = "S3 Error Message";
+
+        // Mock data for downloaded file
+        internal const string S3FileContent = "S3ClientTest";
+
+        // Mock content length
+        internal const int ContentLength = 9999;
+
+        public MockAmazonS3Client(string awsAccessKeyId, string awsSecretAccessKey, string awsSessionToken, AmazonS3Config clientConfig)
+    : base(awsAccessKeyId, awsSecretAccessKey, awsSessionToken, clientConfig)
+        {
+        }
+
+        internal Exception CreateMockAwsResponseError(string errorCode, string isAsync)
```

Comment:
instead of creating a mock class and putting logic into it you could try to use Moq librabary to mock the AmazonS3Client. I mean in the test it would like something like that:
```
var mock = new Mock<AmazonS3Client>();
mock.CallBase = true;
mock.Setup(client => client.GetObjectAsync(...))
         .Returns(something-to-return)
```

so in the test you could write what a given method should return.

Reply:
I refactored the mock class using Moq